### PR TITLE
feat(users): default user

### PR DESCRIPTION
**Pull request by ig-ra**
- move default user to auth middleware
- add config and enable default user in dev and test modes
- fix tests

fixes ENG-1254

### CODE REVIEW

**Review by daabr — APPROVED** [review]

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go [hunk 1]
```diff
@@ -34,19 +36,25 @@ type Deps struct {
...
-
-		if user := authcontext.GetAuthnUser(ctx); !user.IsValid() {
-			if auth := r.Header.Get("Authorization"); auth != "" {
-				kind, payload, _ := strings.Cut(auth, " ")
+		ctx := akCtx.WithRequestOrginator(r.Context(), akCtx.User)
+		user := authcontext.GetAuthnUser(ctx)
+		authHdr := r.Header.Get("Authorization")
 
+		if deps.Cfg.AllowDefaultUser {
+			if user.IsValid() || authHdr != "" {
```
Comment (efiShtain) [inline internal/backend/auth/authhttpmiddleware/authmiddleware.go:46]: Is it possible that the user you got from context is the default user ?
If so, this check would not allow the default user to work

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go [hunk 2]
```diff
@@ -34,19 +36,25 @@ type Deps struct {
...
+				http.Error(w, "only default user is allowed", http.StatusUnauthorized)
+				return
+			}
+			ctx = authcontext.SetAuthnUser(ctx, sdktypes.DefaultUser)
+		} else {
+			if !user.IsValid() && authHdr != "" {
```
Comment (efiShtain) [inline internal/backend/auth/authhttpmiddleware/authmiddleware.go:52]: If the user is not valid and the authHeader is empty, I think you can pass the test and continue.

Isn't it problematic?

Auth header has to be there if users are enabled (default user if off)

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go (hunk 1 above)
Comment (ig-ra) [reply internal/backend/auth/authhttpmiddleware/authmiddleware.go:46]: in my view - default user is not a user that could have token or could be provided externally.
We may consider change it of coz (for example create default credentials file thus providing default user in cli), but meanwhile I see no reason - and backend is handling lack of the user by attaching default user to the request

File: internal/backend/auth/authhttpmiddleware/authmiddleware.go (hunk 2 above)
Comment (ig-ra) [reply internal/backend/auth/authhttpmiddleware/authmiddleware.go:52]: https://linear.app/autokitteh/issue/ENG-1266/auth-middleware-and-no-user

**Pull request merged**